### PR TITLE
Updates ErrorDetail to implement the new IErrorDetail interface found in service-interface-base

### PR DESCRIPTION
**Pull request by hnorrie16**

### CODE REVIEW

File: src/main/java/io/electrum/billpay/model/ErrorDetail.java [hunk 1]
```diff
@@ -215,6 +143,21 @@ public void setDetailMessage(Object detailMessage) {
...
+            validErrorType = true;
+            break;
+         }
+      }
+      if (validErrorType) {
+         this.errorType = ErrorType.valueOf(errorTypeString);
+      } else {
+         throw new Exception("No matching error type for '" + errorTypeString + "' could be found.");
+      }
+   }
```
Comment (HarryGHeathcock) [inline src/main/java/io/electrum/billpay/model/ErrorDetail.java:159]: Needs an @Override tag.

**Review by HarryGHeathcock — COMMENTED** [review]
Some small comments, but mostly done

File: src/main/java/io/electrum/billpay/model/ErrorDetail.java (hunk 1 above)
Comment (stanleydesmond) [reply src/main/java/io/electrum/billpay/model/ErrorDetail.java:159]: This is implementing an abstract method from an interface so I don't think it needs an override ?

Comment (HarryGHeathcock) [reply src/main/java/io/electrum/billpay/model/ErrorDetail.java:160]: It's generally good to use @Override to indicate that it is inherited from somewhere, but in this case it does look it's not complaining about needing those tags, so it should be fine.

**Review by HarryGHeathcock — APPROVED** [review]
Looks good to me

**Pull request merged**